libsplinter/src/service/scabbard/state.rs

```diff
@@ -609,22 +609,44 @@ impl BatchHistory {
 
     fn update_batch_status(&mut self, signature: &str, status: BatchStatus) {
         match self.history.get_mut(signature) {
-            Some(ref mut batch) if batch.status == BatchStatus::Pending => {
-                batch.set_status(status);
+            Some(info) => {
+                if info.status == BatchStatus::Pending {
+                    info.set_status(status);
+                } else {
```

Comment:
This could still have been done as an additional match arm

Reply:
Updated